modtools/item-trigger.lua

```diff
@@ -146,57 +143,105 @@ function compareInvModes(reqMode,itemMode)
  end
 end
 
-function checkMode(table,triggerArg)
- for _,command in ipairs(triggerArg) do
-  if command[""..table.mode..""] then
-   local reqModeType = command[""..table.mode..""]
-   local modeType = tonumber(table.modeType)
+function checkMode(triggerArgs,table)
+ local mode = table.mode
+ for _,argArray in ipairs(triggerArgs) do
+  if argArray[""..mode..""] then
```

Comment:
`tostring(mode)` or `""..mode` should work

Reply:
Changed.